src/Symmetric.jl

```diff
@@ -0,0 +1,213 @@
+@doc doc"""
+    SymmetricMatrices{N,T} <: Manifold
+
+The [`Manifold`](@ref) $ Sym (N)$ consisting of the `T`-valued, symmetric matrices of size $ n\times n$, i.e. the set 
+
+$\operatorname{Sym}(n) = \bigl\{A \in \mathbb F^{n\times n} \big| A^{\mathrm{T}} = A \bigr\}$
+
+where $\mathbb F \in \{ \mathbb R, \mathbb C\}$.
+For `T`, there is a distinction between 
+* `T = Real`: real-valued matrix entries
+* `T = Complex`: complex-valued matrix entries
+
+Being slightly inefficient, the matrices are safed as $n\times n$ arrays despite the redundance of the lower triangle.
+
+
+# Constructor
+
+    SymmetricMatrices(n::Int,T::Type = Real)
+
+"""
+struct SymmetricMatrices{N,T} <: Manifold end
+SymmetricMatrices(N::Int,T::Type = Real) = SymmetricMatrices{N,T}()
+
+function representation_size(::SymmetricMatrices{N,T}) where {N,T}
+    return (N,N)
+end
+
+@doc doc"""
+    manifold_dimension(M::SymmetricMatrices{n,T}) where {n,T <: Real}
+
+Return the dimension of the [`SymmetricMatrices`](@ref) matrix `M` with real-valued entries, i.e. 
+````math
+    \frac{n(n+1)}{2}
+````
+"""
+manifold_dimension(M::SymmetricMatrices{N,Real}) where {N,T} = div(N*(N+1),2)
+
+@doc doc"""
+    manifold_dimension(M::SymmetricMatrices{n,T}) where {n,T <: Complex}
+
+Return the dimension of the [`SymmetricMatrices`](@ref) matrix `M` with complex-valued entries, i.e. 
+````math
+    n(n+1)
+````
+"""
+manifold_dimension(M::SymmetricMatrices{N,Complex}) where {N,T} = N*(N+1)
+
+@doc doc"""
+    project_point!(M::SymmetricMatrices,x)
+    
+Implements the [`project_point`](@ref project_point!(M::Manifold, x)) of an arbitrary matrix `x` onto the manifold of [`SymmetricMatrices`](@ref), i.e. 
+
+$x_{new} = \frac{1}{2} \left( x_{old} + x_{old}^{\mathrm{T}} \right).$
```

Comment:
For consistency this could also be set with ` ````math ` and ` ```` ` (where you don't have to indent internally I think).